lib/mpl_toolkits/axes_grid1/axes_size.py

```diff
@@ -162,75 +162,40 @@ def add_artist(self, a):
 
     def get_size(self, renderer):
         rel_size = 0.
-        w_list, h_list = [], []
-        for a in self._artist_list:
-            bb = a.get_window_extent(renderer)
-            w_list.append(bb.width)
-            h_list.append(bb.height)
-        dpi = a.get_figure().get_dpi()
-        if self._w_or_h == "width":
-            abs_size = max(w_list)/dpi
-        elif self._w_or_h == "height":
-            abs_size = max(h_list)/dpi
-
+        extent_list = [
+            getattr(a.get_window_extent(renderer), self._w_or_h) / a.figure.dpi
+            for a in self._artist_list]
+        abs_size = max(extent_list, default=0)
         return rel_size, abs_size
 
 
-class MaxWidth(_Base):
+class MaxWidth(MaxExtent):
     """
-    Size whose absolute part is the largest width of
-    the given *artist_list*.
+    Size whose absolute part is the largest width of the given *artist_list*.
     """
-    def __init__(self, artist_list):
-        self._artist_list = artist_list
-
-    def add_artist(self, a):
-        self._artist_list.append(a)
 
-    def get_size(self, renderer):
-        rel_size = 0.
-        w_list = []
-        for a in self._artist_list:
-            bb = a.get_window_extent(renderer)
-            w_list.append(bb.width)
-        dpi = a.get_figure().get_dpi()
-        abs_size = max(w_list)/dpi
-
-        return rel_size, abs_size
+    def __init__(self, artist_list):
+        super().__init__(artist_list, "width")
 
 
 class MaxHeight(_Base):
```

Comment:
I think this should be:
```suggestion
class MaxHeight(MaxExtent):
```